pkg/virt-operator/resource/apply/patches.go

```diff
@@ -69,6 +70,23 @@ func addFlagsPatch(name, resource string, flags map[string]string, patches []v1.
 	})
 }
 
+func addKubeletPodsDirPatch(name, resource string, flags map[string]string, patches []v1.CustomizeComponentsPatch) []v1.CustomizeComponentsPatch {
+	if len(flags) == 0 {
+		if _, ok := flags["kubelet-pods-dir"]; !ok {
+			return patches
+		}
+	}
+	kubeletPodsDir := flags["kubelet-pods-dir"]
+	return append(patches, v1.CustomizeComponentsPatch{
+		ResourceName: name,
+		ResourceType: resource,
+		Patch: fmt.Sprintf(`{"spec":{"template":{"spec":{"containers":[{"name":"%s","volumeMounts":[{"mountPath":"%s","mountPropagation":"Bidirectional","name":"kubelet-pods"}]}],"volumes":[{"hostPath":{"path":"%s"},"name":"kubelet-pods-shortened"},{"hostPath":{"path":"%s"},"name":"kubelet-pods"}]}}}}`,
+			name, kubeletPodsDir, kubeletPodsDir, kubeletPodsDir),
+		//Patch:        fmt.Sprintf(`{"spec":{"template":{"spec":{"containers":[{"name":%q,"command":["%s","%s"]}]}}}}`, name, name, strings.Join(flagsToArray(flags), `","`)),
+		Type: v1.StrategicMergePatchType,
+	})
+}
+
```

Comment:
I don't think this is necessary. It should be possible for you to configure what you're wanting today  using the `kubevirt.spec.customizeComponents.patches` field. This is reason that field exists.

Reply:
Also maybe I am missing something but we already handle flags for the handler so whoever wants to change the kubelet path it should be doable as well.

Reply:
Thank you for your attention, `addFlagsPatch` method will do change flag in virt-handler. While if we have k8s with non-default kubelet pod dir and want make virt-handler running by edit `kubelet path` flag for handler in kubevirt cr. We also need change `mountPath` in handler ds. So we need `addKubeletPodsDirPatch` to do this.

Reply:
I think combination of `customizeComponents.patches` and `flags` is possible here. We should not expand the API too much.

Reply:
I understand what you are talking about. We can maunally add patch for virt-handler ds with `customizeComponents.patches`.

Reply:
Have you considered adding parameters in kubevirt cr or operator yaml to specify values ​​like kubelet pod dir so that we don't need the patch after all handler pod has ready?

Reply:
> Have you considered adding parameters in kubevirt cr or operator yaml to specify values ​​like kubelet pod dir so that we don't need the patch after all handler pod has ready?

You should be able to create the Kubevirt CR before deployment. Let us know if that is not the case